src/Catty/ViewController/Download/StoreProjectDownloader.swift

```diff
@@ -183,3 +185,41 @@ enum ProjectType {
     case mostViewed
     case mostRecent
 }
+
+extension StoreProjectDownloader {
+    func download(project: StoreProject, completion: @escaping (Data?, StoreProjectDownloaderError?) -> Void, progression: ((Float) -> Void)?) {
+        guard let indexURL = URL(string: "https://share.catrob.at/app/download/\(project.projectId).catrobat") else { return }
```

Comment:
Please use `NetworkDefines.downloadUrl` instead of the direct URL `https://share.catrob.at/app/download/`.

Reply:
I have tried `NetworkDefines.downloadUrl` but wasn't able to get any data during testing which uses  `https://web-test.catrob.at/pocketcode/`, I tried to `curl `the link `https://web-test.catrob.at/pocketcode/821.catrobat`, it replies back with 404. That's why I used the production URL.

Can you suggest anything?

Reply:
That is because the development environment has a different database than the production environment and a project with 821 did not exist there. You can download another project there, e.g. _https://web-test.catrob.at/pocketcode/download/00fbfbe1-6f87-11ea-959c-000c293d4b76.catrobat_

Reply:
So, shall I create a new StoreProject object with id `00fbfbe1-6f87-11ea-959c-000c293d4b76` and use that in my tests?

Reply:
In your test fixture or where do you want to do that? If it is in the test fixture, then there is no need to do this (the information in the fixture is not really relevant as long at is has the right structure).

Reply:
Now I understand. You are making real requests at `testDownloadProjectProgression`. I'll leave some suggestions below (basically we should not make real API requests at this level of our test suite).

Reply:
I think except `projectId`  all other information is not relevant.

so, shall I change the  function `private func getStoreProjectMock() -> StoreProject` to ` private func getStoreProjectMock(withProjectId id: String) -> StoreProject`  which will let us to modify the `projectId` without changing other informations.

Reply:
> I think except projectId all other information is not relevant.

True. Please change the first parameter to `projectId: String`.

Reply:
Sorry for the confusion. I was talking about the parameter of this `download` method, not the `StoreProjectMock`. The `StoreProject` is a DTO which contains a lot of data (e.g. number of downloads, screenshots, etc.). This data is not necessary here and often also not available. So, since you are right that we only need the projectId for this method, I think it will be better to just use this parameter here (`download(projectId: String, ...)`). I will hand in a request to do the same for `fetchProjectDetails` in the future.